src/test/netbase_tests.cpp

```diff
@@ -576,4 +576,24 @@ BOOST_AUTO_TEST_CASE(caddress_unserialize_v2)
     BOOST_CHECK(fixture_addresses == addresses_unserialized);
 }
 
+BOOST_AUTO_TEST_CASE(isbadport)
+{
+    BOOST_CHECK(IsBadPort(1));
+    BOOST_CHECK(IsBadPort(22));
+    BOOST_CHECK(IsBadPort(6000));
+
+    BOOST_CHECK(!IsBadPort(80));
+    BOOST_CHECK(!IsBadPort(443));
+    BOOST_CHECK(!IsBadPort(8333));
+
+    // Check all ports, there must be 80 bad ports in total.
+    size_t total_bad_ports{0};
+    for (uint16_t port = std::numeric_limits<uint16_t>::max(); port > 0; --port) {
```

Comment:
634f2838ee137234093a1fc89162f395366a387f might be a good place to use `std::count_if`, or alternatively, possibly faster given the number of iterations with
```suggestion
    for (uint16_t port = std::numeric_limits<uint16_t>::max(); port != 0; --port) {
```

Reply:
`std::count_if` works with iterators, how could I use it to go over the numbers 1..64k? Are you saying `!=` is faster than `>`? Even if "yes", I would leave it as is coz I find the `>` more natural/readable.

Reply:
Right, and `std::ranges::count_if` is C++20 (and test code doesn't need to be elegant).

`!= 0` might be a faster iterator comparator (e.g. https://www.youtube.com/watch?v=oelQ4uAw2WQ) but in this case of a "manual" loop it may also be inlined by compilers in optimized non-debug modes. A non-blocking thought in any case.

Reply:
Huh, just rewatched the video and it seems `!=` was also what a ranged for-loop generated (GCC 10.2 -O3). Ofc feel free to ignore.